edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.MqttBrokerAdapter/brokerConnection/MqttBrokerConnector.cs

```diff
@@ -32,10 +32,14 @@ public class MqttBrokerConnector : IMqttBrokerConnector
 
         AtomicBoolean isRetrying = new AtomicBoolean(false);
 
+        public TimeSpan AckTimeout { get; set; }
```

Comment:
Pls make it read only

Reply:
This value needs to be modified after instantiation, that is the reason this property is not read-only. The full story:
- The ack timeout was added to avoid that very slim possibility that on a local connection the broker (or the mqtt library) does not ack back a publish operation, and the call get stuck forever. The very first implementation used a 10 sec hardcoded value.
- Because of that the related unit test needed to wait for this 10 sec to expire. Reacting to the PR feedbacks, I added this property, so the unit test can lower the value.

Other options could have been:
- add the ack timeout via constructor parameter. I did not do that for two reasons:
  - 'too visible' change, considering that it is only to make a test run shorter
  - passing numeric values in constructor is not too autofac friendly. Currently this class does not have a custom activator and creating one just for the sake of a faster unit test did not seem reasonable.
- make the originally constant value a read-only private variable and modify it from the unit test via reflection. Although I liked that idea because it would have left the class interface clean, according to previous experiences, tricks like that are not always popular.
- So I ended up adding this property which can be used after the class is instantiated, so the unit test can change the timeout before it publishes. The drawback that it is still to 'visible', when the only reason is to speed up a test.
- (also considered creating the unit test only for myself to see that the code was working, then disable with a skip="some explanation")